src/main/java/com/google/jenkins/plugins/k8sengine/VerificationTask.java

```diff
@@ -0,0 +1,128 @@
+ * to the KubectlWrapper for querying the kubernetes cluster.
+ */
+public class VerificationTask {
+  private static final int VERIFICATION_DELAY = 5;
+
+  private KubectlWrapper kubectl;
+  private Manifests.ManifestObject manifestObject;
+  private KubernetesVerifiers.VerificationResult currentResult;
+  private PrintStream consoleLogger; // Jenkins console
+
+  private static Logger LOGGER = Logger.getLogger(VerificationTask.class.getName());
+
+  /**
+   * @param kubectl The KubectlWrapper for issuing kubectl commands.
+   * @param manifestObject The wrapper for the kubernetes object ot verify.
+   * @param consoleLogger The Console output PrintStream.
+   */
+  public VerificationTask(
+      KubectlWrapper kubectl, Manifests.ManifestObject manifestObject, PrintStream consoleLogger) {
+    this.kubectl = kubectl;
+    this.manifestObject = manifestObject;
+    this.consoleLogger = consoleLogger;
+  }
+
+  /** @return true if this task has successfully verified the manifestObject */
+  public boolean isVerified() {
+    return currentResult != null && currentResult.isVerified();
+  }
+  /** @return the VerificationResult */
+  public KubernetesVerifiers.VerificationResult getVerificationResult() {
+    return currentResult;
+  }
+
+  /**
+   * Verifies it's manifestObject using the kubectl cli
+   *
+   * @return this after performing verify
+   */
+  public VerificationTask verify() {
+    consoleLogger.println(String.format("Verifying: %s ", manifestObject.describe()));
+    currentResult = KubernetesVerifiers.verify(kubectl, manifestObject);
+    if (isVerified()) {
+      consoleLogger.println(currentResult.toString());
+    }
+    return this;
+  }
+  /**
+   * @param kubectl KubectlWrapper object for issuing commands to kubernetes cluster
+   * @param manifestObjects list of manifest objects to verify
+   * @param consoleLogger PrintStream for outputting results (intended to be user facing)
+   * @param timeoutInMinutes Stop retrying verification after this many minutes
+   */
+  public static boolean verifyObjects(
+      @Nonnull KubectlWrapper kubectl,
+      @Nonnull List<Manifests.ManifestObject> manifestObjects,
+      @Nonnull PrintStream consoleLogger,
+      int timeoutInMinutes) {
+
+    List<VerificationTask> verificationTasks =
+        manifestObjects
+            .stream()
+            .map((manifestObject) -> new VerificationTask(kubectl, manifestObject, consoleLogger))
+            .collect(Collectors.toList());
+
+    Repeat.onlyIf(
+            (ctx) ->
+                !verificationTasks
+                    .stream()
+                    .map((task) -> task.isVerified()) // only repeat if we aren't all done
+                    .reduce(true, (acc, done) -> acc && done))
+        .backoff(Backoff.fixed(Duration.ofSeconds(VERIFICATION_DELAY)))
+        .timeout(Duration.ofMinutes(timeoutInMinutes))
+        .apply((Publisher<VerificationTask>) Flux.fromIterable(verificationTasks))
```

Comment:
I see what you meant by dense here. Could you add a comment on this step as well? As my first time dealing with Reactor I'm not too sure what this `apply` does.

Reply:
apply applies the Repeat(er) to the Flux (the event stream data structure) I've added that comment for next push.